### PR TITLE
add support for python 3.11

### PR DESCRIPTION
**Pull request by GrozniyToaster**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,16 +38,18 @@ authors = [
...
 
 [tool.poetry.dependencies]
-python = ">=3.8.0, <3.11.0"
+python = ">=3.8.0, <3.12.0"
 scikit-learn = ">=0.24, <2"
 pandas = "^1.1"
 catboost = ">=0.21"
 ruptures = "^1.1.5"
-numba = ">=0.53.1,<0.56.0"
+numba = ">=0.53.1,<0.58.0"
```
Comment (Mr-Geekman) [inline pyproject.toml:52]: Why do we need this change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -
 - Class `Tune` for hyperparameter optimization within existing pipeline ([#1200](https://github.com/tinkoff-ai/etna/pull/1200))
 - Add `etna.distributions` for using it instead of using `optuna.distributions` ([#1292](https://github.com/tinkoff-ai/etna/pull/1292))
+- Add support python 3.11 ([#1301](https://github.com/tinkoff-ai/etna/pull/1301))
```
Comment (Mr-Geekman) [inline CHANGELOG.md:19]: Have you tried to rerun all the tests on python 3.11?

Comment (Mr-Geekman) [reply CHANGELOG.md:19]: Shouldn't we also add CI-job with python 3.11?

Comment (GrozniyToaster) [reply CHANGELOG.md:19]: I'll add

File: pyproject.toml (hunk 1 above)
Comment (GrozniyToaster) [reply pyproject.toml:52]: since version 0.57 python 3.11 is supported

**Comment by GrozniyToaster** [comment]
It seems that due to pytorch-forecasting it will not be possible to add 3.11 to the library yet

https://pypi.org/project/pytorch-forecasting/

**Pull request closed without merging**